src/routes/messages/messages.service.ts

```diff
@@ -99,8 +97,24 @@ export class MessagesService {
       ),
     );
 
-    return sortBy(Object.entries(groups), ([timestamp]) => timestamp).map(
-      ([timestamp, messages]) => [Number(timestamp), messages],
+    return (
+      Object.keys(groups)
+        // We first sort the groups in descending order (most recent first)
+        .sort((day1, day2) => day2.localeCompare(day1))
+        // For each group we create a tuple of the timestamp of the day
+        // with the messages for that day sorted by creation in descending order
+        .map((groupKey) => {
+          return [
+            Number(groupKey),
+            orderBy(
+              groups[groupKey],
+              (message) => {
+                message.created.getTime();
+              },
+              ['desc'],
+            ),
+          ];
+        })
```

Comment:
`orderBy` isn't changing anything here (nothing is being returned in [the iteratee](https://lodash.com/docs/4.17.15#orderBy)). It can be removed without breaking the tests.

```suggestion
        .map((groupKey) => {
          return [
            Number(groupKey),
            groups[groupKey]
          ];
        })
```

Reply:
You are right. It was not returning from the function and it only worked because the `messages` were in the correct order.

I've added the return and adjusted the tests to include different hours (for the messages on the same day) [0a7753d](https://github.com/safe-global/safe-client-gateway-nest/pull/363/commits/0a7753d8541a2a45eb8d72fa29fb92af0e0efb93)